### ISSUE TITLE
Tailoring seems crushing the game from time to time

### ISSUE THREAD

**Issue by dranitski** [CleverRaven/Cataclysm-DDA#51828]
### Describe the bug

Tailoring activity crashes the game

### Steps To Reproduce

1. create new world without any mods, except these game suggests by default (bionic prof, no fungi growth, etc.)
2. create random char
3. give yourself atomic lamp (or other source of light), tailors kit, 1000 thread, 100 cotton patchwork cloth parts
4. start using your tailors kit on any piece of cloth, for example on these you wear
5. in tailoring menu choose "repeat until reinforced"
6. try different cloths. For me it always fails on "long-sleeving shirt" and "pair of socks" that my player is wearing
7. Enjoy the crash
```
The program has crashed.
See the log file for a stack trace.
CRASH LOG FILE: /home/dzmitry-dranitski/.cataclysm-dda/config/crash.log
VERSION: 4f8529e
TYPE: Signal
MESSAGE: SIGSEGV: Segmentation faultError creating SDL message box: No message system available

STACK TRACE:

    ./cataclysm-tiles(debug_write_backtrace(std::ostream&)+0x41) [0x55761dc7a4c3]
    ./cataclysm-tiles(+0x96304e) [0x55761dc5c04e]
    ./cataclysm-tiles(+0x963296) [0x55761dc5c296]
    /usr/lib/libc.so.6(+0x3cda0) [0x7fa1af0a5da0]
    ./cataclysm-tiles(item::has_flag(string_id<json_flag> const&) const+0x7c) [0x55761de99058]
    ./cataclysm-tiles(activity_handlers::repair_item_finish(player_activity*, Character*)+0x39c) [0x55761da35e42]
    ./cataclysm-tiles(std::function<void (player_activity*, Character*)>::operator()(player_activity*, Character*) const+0x24) [0x55761da5dbd8]
    ./cataclysm-tiles(activity_type::call_finish(player_activity*, Character*) const+0x37) [0x55761da5cd23]
    ./cataclysm-tiles(player_activity::do_turn(Character&)+0x95e) [0x55761e3d09fa]
    ./cataclysm-tiles(do_turn()+0x378) [0x55761dcc1c1a]
    ./cataclysm-tiles(main+0x12a8) [0x55761d8bcddc]
    /usr/lib/libc.so.6(__libc_start_main+0xd5) [0x7fa1af090b25]
    ./cataclysm-tiles(_start+0x2e) [0x55761d9d46fe]

    Attempting to repeat stack trace using debug symbols…
    debug_write_backtrace(std::ostream&)
    ??:?
    log_crash
    crash.cpp:?
    signal_handler
    crash.cpp:?
    ??
    ??:0
    item::has_flag(string_id<json_flag> const&) const
    ??:?
    activity_handlers::repair_item_finish(player_activity*, Character*)
    ??:?
    std::function<void (player_activity*, Character*)>::operator()(player_activity*, Character*) const
    ??:?
    activity_type::call_finish(player_activity*, Character*) const
    ??:?
    player_activity::do_turn(Character&)
    ??:?
    do_turn()
    ??:?
    main
    ??:?
    __libc_start_main
    ??:?
    _start
    ??:?
Aborted (core dumped)
15:59:20-dzmitry.dranitski@:~/repos/Cataclysm-DDA$ ./cataclysm-tiles 
The program has crashed.
See the log file for a stack trace.
CRASH LOG FILE: /home/dzmitry-dranitski/.cataclysm-dda/config/crash.log
VERSION: 4f8529e
TYPE: Signal
MESSAGE: SIGSEGV: Segmentation faultError creating SDL message box: No message system available

STACK TRACE:

    ./cataclysm-tiles(debug_write_backtrace(std::ostream&)+0x41) [0x55cc81aec4c3]
    ./cataclysm-tiles(+0x96304e) [0x55cc81ace04e]
    ./cataclysm-tiles(+0x963296) [0x55cc81ace296]
    /usr/lib/libc.so.6(+0x3cda0) [0x7f773f5b9da0]
    ./cataclysm-tiles(item::has_flag(string_id<json_flag> const&) const+0x7c) [0x55cc81d0b058]
    ./cataclysm-tiles(activity_handlers::repair_item_finish(player_activity*, Character*)+0x39c) [0x55cc818a7e42]
    ./cataclysm-tiles(std::function<void (player_activity*, Character*)>::operator()(player_activity*, Character*) const+0x24) [0x55cc818cfbd8]
    ./cataclysm-tiles(activity_type::call_finish(player_activity*, Character*) const+0x37) [0x55cc818ced23]
    ./cataclysm-tiles(player_activity::do_turn(Character&)+0x95e) [0x55cc822429fa]
    ./cataclysm-tiles(do_turn()+0x378) [0x55cc81b33c1a]
    ./cataclysm-tiles(main+0x12a8) [0x55cc8172eddc]
    /usr/lib/libc.so.6(__libc_start_main+0xd5) [0x7f773f5a4b25]
    ./cataclysm-tiles(_start+0x2e) [0x55cc818466fe]

    Attempting to repeat stack trace using debug symbols…
    debug_write_backtrace(std::ostream&)
    ??:?
    log_crash
    crash.cpp:?
    signal_handler
    crash.cpp:?
    ??
    ??:0
    item::has_flag(string_id<json_flag> const&) const
    ??:?
    activity_handlers::repair_item_finish(player_activity*, Character*)
    ??:?
    std::function<void (player_activity*, Character*)>::operator()(player_activity*, Character*) const
    ??:?
    activity_type::call_finish(player_activity*, Character*) const
    ??:?
    player_activity::do_turn(Character&)
    ??:?
    do_turn()
    ??:?
    main
    ??:?
    __libc_start_main
    ??:?
    _start
    ??:?
Aborted (core dumped)
```

### Expected behavior

No crash

### Versions and configuration

OS: Arch Linux
Experimental https://github.com/CleverRaven/Cataclysm-DDA/commit/4f8529e53d1462ae7404cd61c125a4bc1190c383

### Additional context

I've prepared a savegame of my world to reproduce. However ive faced it in different worlds

[tailoring.zip](https://github.com/CleverRaven/Cataclysm-DDA/files/7217799/tailoring.zip)

**Comment by atealein** [comment]
Alright, just as an additional confirmation, all repair actions can cause it - with tailoring seems to happen more often on reinforce, while repairing with soldering iron seems to triger it even on "Repeat once" the repair action. 

It is definitely something that broke between 15/9 and 23/9 as my old version ( 6725ecf ) didn't have this issue and the new version ( e1e5c9e ) does. 

I had a bit of experiment now, because I updated my version, played a bit, noticed my rucksack and my atomic headlight are a bit broken, so I tried to fix them. It caused immediate "Segmentation fault (core dumped)" crash on trying to repair using my soldering iron, the rucksack I was able to repair with tailoring until full, trying to reinforce it triggered the same crash. I returned the save to my old version of the game, load it up, repaired the items without any issues.
